futurenhs.app/components/ContentBlockManager/index.tsx

```diff
@@ -0,0 +1,688 @@
+    }
+
+    /**
+     * Action buttons
+     */
+    const EnterUpdateButton: ({ isDisabled: boolean }) => JSX.Element = ({ isDisabled }) => <button disabled={isDisabled} className={generatedClasses.headerCallOutButton} onClick={handleSetToUpdateMode}>{headerEnterUpdateButton}</button>
+    const LeaveUpdateButton: ({ isDisabled: boolean }) => JSX.Element = ({ isDisabled }) => <button disabled={isDisabled} className={generatedClasses.headerCallOutButton} onClick={handleSetToReadMode}>{headerLeaveUpdateButton}</button>
+    const DiscardUpdateButton: ({ isDisabled: boolean }) => JSX.Element = ({ isDisabled }) => <button disabled={isDisabled} className={generatedClasses.headerCallOutButton} onClick={handleDiscardUpdates}>{headerDiscardUpdateButton}</button>
+    const PreviewUpdateButton: ({ isDisabled: boolean }) => JSX.Element = ({ isDisabled }) => <button disabled={isDisabled} className={generatedClasses.headerCallOutButton} onClick={handleSetToPreviewMode}>{headerPreviewUpdateButton}</button>
+    const PublishUpdateButton: ({ isDisabled: boolean }) => JSX.Element = ({ isDisabled }) => <button disabled={isDisabled} className={generatedClasses.headerPrimaryCallOutButton} onClick={handleUpdateBlockSubmit}>{headerPublishUpdateButton}</button>
+
+    /**
+     * Handle creating a new block instance from the page template and adding it to the active block list
+     */
+    const handleCreateBlock = (instanceId: string): void => {
+
+        const updatedBlocks = [...blocks];
+        const block = simpleClone(blocksTemplate[instanceId]);
+
+        block.instanceId = createHtmlSafeId();
+        updatedBlocks.push(block)
+
+        handleSetToUpdateMode();
+        setBlocks(updatedBlocks);
+        blocksChangeAction?.(updatedBlocks);
+
+    }
+
+    /**
+     * Handle deleting a block instance from the active block list
+     */
+    const handleDeleteBlock = (instanceId: string): void => {
+
+        const index: number = blocks.findIndex((block) => block.instanceId === instanceId);
+        const updatedBlocks = deleteArrayItem(blocks, index);
+
+        if (localErrors.current[instanceId]) {
+
+            delete localErrors.current[instanceId];
+
+        }
+
+        setBlocks(updatedBlocks);
+        blocksChangeAction?.(updatedBlocks);
+
+    }
+
+    /**
+     * Handle moving a block instance one index backwards in the active block list
+     */
+    const handleMoveBlockPrevious = (instanceId: string): void => {
+
+        const index: number = blocks.findIndex((block) => block.instanceId === instanceId);
+        const targetIndex: number = index - 1;
+        const updatedBlocks = moveArrayItem(blocks, index, targetIndex);
+
+        setBlocks(updatedBlocks);
+        blocksChangeAction?.(updatedBlocks);
+
+        setTimeout(() => {
+
+            const targetSelector: string = updatedBlocks[targetIndex].instanceId;
+
+            document.getElementById(targetSelector)?.focus()
+
+        }, 0)
+
+    }
+
```

Comment:
Maybe one for a refactor later but I feel like the previous/next could maybe be one function and whether the targetIndex is index -1 or +1 could be controlled by another argument?